rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -223,6 +227,16 @@ func (s *RollupSyncService) fetchRollupEvents() error {
 
 				return nil
 			}
+			if errors.Is(err, ErrMissingBatchEvent) {
+				// If there's a missing batch event, rewind the L1 sync height by some blocks to re-fetch from L1 RPC and
+				// replay creating corresponding CommittedBatchMeta in local DB.
+				// This happens recursively until the missing event has been recovered as we will call fetchRollupEvents again
+				// with the `L1Height = prevL1Height - rewindL1Height`.
+				s.callDataBlobSource.SetL1Height(prevL1Height - rewindL1Height)
+
+				return fmt.Errorf("missing batch event, rewinding L1 sync height by %d blocks: %w", rewindL1Height, err)
```

Comment:
Might be useful to log the height we rewinded to

Reply:
done